src/main/java/org/cactoos/text/JoinedText.java

```diff
@@ -34,18 +35,11 @@
  * <p>There is no thread-safety guarantee.
  *
  * @since 0.9
+ * @todo #849:30min Continue refactoring all classes implementing Text to extend
+ *  TextEnvelope - in most cases asString should be removed and implementation
+ *  from TextEnvelope should be used.
```

Comment:
@krzyk I don't think that removing `asString` is a good idea; I think we should maintain it as it is and rely on `equals` and `hashCode` for comparisons, and override these methods methods if we need some specific way of comparing these objects. For example, `new LowerText("UPPERTEXT").equals("uppertext")` should return `true`. WDYT?

Reply:
@paulodamaso ok, we should compare using equals, but I don't see how it is relevant to removing `asString` from some implementations where it is exactly the same as in the superclass - `TextEnvelop`.

Reply:
@krzyk You are right, using lambdas in constructor makes these methods just code duplication, see below.